packages/http/src/mocker/HttpMocker.ts

```diff
@@ -32,48 +40,68 @@ export class HttpMocker implements IMocker<IHttpOperation, IHttpRequest, IHttpCo
       throw new Error('Http request is not defined');
     }
 
-    // setting default values
-    const acceptMediaType = input.data.headers && caseless(input.data.headers).get('accept');
-    config = config || { mock: false };
-    const mockConfig: IHttpOperationConfig =
-      config.mock === false ? { dynamic: false } : Object.assign({}, config.mock);
+    return new Reader<Logger, IHttpOperationConfig>(logger => {
```

Comment:
Does this feel a bit invasive to you? I am not sure about it. Do we need to embed our code in a function just to get logger doing its job?

Reply:
I'm resolving this because the code has been refactored.